src/modify.js

```diff
@@ -80,10 +80,22 @@ function rewriteAllFields(schema, configCallback, context) {
 
 export function modify(originalSchema, config) {
   const schema = JSON.parse(JSON.stringify(originalSchema));
+  let warn = []; // To be implemented in next PRs.
 
+  // All these functions mutate "schema",
+  // that's why we create a copy above
   rewriteFields(schema, config.fields);
 
   rewriteAllFields(schema, config.allFields);
 
-  return schema;
```

Comment:
I decided to create a specific PR for this breaking change. Once merged, all the the other open PRs (order, pick, create) are easier to work on top of.

The new version (10.1.0) will only be published when all the other 3 PRs are approved.

Does that make sense to you?

Reply:
Makes sense @sandrina-p, thanks for splitting this for easy reviewing!!